uqcsbot/dominos_coupons.py

```diff
@@ -44,50 +48,105 @@ async def dominoscoupons(
         number_of_coupons: int = 5,
         ignore_expiry: bool = True,
         keywords: str = "",
+        source: Literal["both", "couponese", "frugalfeeds"] = "both",
     ):
         """
         Returns a list of dominos coupons
         """
+
         if number_of_coupons < 1 or number_of_coupons > MAX_COUPONS:
             await interaction.response.send_message(
                 content=f"You can't have that many coupons. Try a number between 1 and {MAX_COUPONS}.",
                 ephemeral=True,
             )
             return
+
+        if source != "both":
+            if source != "couponese" and source != "frugalfeeds":
+                await interaction.response.send_message(
+                    content=f"That website isn't recognised. Try couponese or frugalfeeds.",
+                    ephemeral=True,
+                )
+                return
+
         await interaction.response.defer(thinking=True)
 
-        try:
-            coupons = _get_coupons(number_of_coupons, ignore_expiry, keywords.split())
-        except RequestException as error:
-            resp_content = (
-                error.response.content if error.response else "No response error given."
-            )
-            logging.warning(
-                f"Could not connect to dominos coupon site ({COUPONESE_DOMINOS_URL}): {resp_content}"
-            )
-            await interaction.edit_original_response(
-                content=f"Sadly could not reach the coupon website (<{COUPONESE_DOMINOS_URL}>)..."
-            )
-            return
-        except HTTPResponseException as error:
-            logging.warning(
-                f"Received a HTTP response code {error.http_code}. Error information: {error}"
-            )
-            await interaction.edit_original_response(
-                content=f"Could not find the coupons on the coupon website (<{COUPONESE_DOMINOS_URL}>)..."
-            )
-            return
+        coupons: List[Coupon] = []
+        stored_url: str = " "
+
+        for tries in range(0, NUMBER_WEBSITES):
```

Comment:
Instead of looping over a range, it might be better to loop over a list of the URLs you are going to gather coupons from. You appear to do this within `_get_coupons_from_page`,  so I'm unsure why multiple attempts are needed here.

Reply:
I really like this idea (I admit my implementation could be better), as well as the idea you left below in the unresolved comment. I'll begin working on this, and the code should be a lot cleaner afterwards.